### PR TITLE
Fix dropout issue in swin transformers

### LINKED ISSUE #7103: Dropout in SwinTransformer unable to set training=False
### 🐛 Describe the bug

https://github.com/pytorch/vision/blob/93df9a50885d0345e31bba691576c83d5cee7737/torchvision/models/swin_transformer.py#L209-L214

`training` argument is not used here, so `training=True` (by default) even when `model.training=False`.

### Versions

Collecting environment information...
PyTorch version: 1.13.1+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.1 LTS (x86_64)
GCC version: (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Clang version: Could not collect
CMake version: version 3.22.4
Libc version: glibc-2.35

Python version: 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0] (64-bit runtime)
Python platform: Linux-5.15.0-56-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3070 Ti
Nvidia driver version: 525.60.13
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.7.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.24.1
[pip3] torch==1.13.1
[pip3] torchaudio==0.13.1
[pip3] torchtext==0.14.1
[pip3] torchvision==0.14.1
[conda] Could not collect

### PR DESCRIPTION
**Pull request by oke-aditya**
Closes #7103

### CODE REVIEW

File: torchvision/models/video/swin_transformer.py [hunk 1]
```diff
@@ -124,6 +124,7 @@ def shifted_window_attention_3d(
     dropout: float = 0.0,
     qkv_bias: Optional[Tensor] = None,
     proj_bias: Optional[Tensor] = None,
+    training: bool = False,
```
Comment (NicolasHug) [inline torchvision/models/video/swin_transformer.py:127]: Should we set the default to True since it's the default of `torch.nn.functional.dropout`, and also the default of `shifted_window_attention` ?

File: torchvision/models/swin_transformer.py [hunk 2]
```diff
@@ -126,7 +126,8 @@ def shifted_window_attention(
     qkv_bias: Optional[Tensor] = None,
     proj_bias: Optional[Tensor] = None,
     logit_scale: Optional[torch.Tensor] = None,
-):
+    training: bool = True,
```
Comment (NicolasHug) [inline torchvision/models/swin_transformer.py:129]: It is a bit sad that we can't put this parameter closer to the `dropout` ones (that would break BC).
Let's add a comment on top of `attention_dropout` saying something like 

```
# If you set the dropout parameters, don't forget to also set the training parameter below.
```

**Review by NicolasHug — COMMENTED** [review]
Thanks a lot for the PR @oke-aditya !
Some minor suggestions below, LMK what you think.

File: torchvision/models/video/swin_transformer.py (hunk 1 above)
Comment (oke-aditya) [reply torchvision/models/video/swin_transformer.py:127]: Well caught this was mistake it should have been True

File: torchvision/models/swin_transformer.py (hunk 2 above)
Comment (oke-aditya) [reply torchvision/models/swin_transformer.py:129]: I have added docstring for training. That would help

**Review by NicolasHug — APPROVED** [review]
Thanks a lot @oke-aditya , LGTM!
